test/Identity.IntegrationTest/Endpoints/IdentityServerTests.cs

```diff
@@ -0,0 +1,455 @@
+                { "client_secret", "something" },
+                { "scope", "api.organization" },
+            }));
+
+            Assert.Equal(StatusCodes.Status400BadRequest, context.Response.StatusCode);
+
+            var errorBody = await AssertHelper.AssertResponseTypeIsAsync<JsonDocument>(context);
+            var error = AssertHelper.AssertJsonProperty(errorBody.RootElement, "error", JsonValueKind.String).GetString();
+            Assert.Equal("invalid_client", error);
+        }
+
+        /// <summary>
+        /// This test currently does not test any code that is not covered by other tests but 
+        /// it shows that we probably have some dead code in <see cref="Core.IdentityServer.ClientStore"/>
+        /// for installation, organization, and user they split on a <c>'.'</c> but have already checked that at least one
+        /// <c>'.'</c> exists in the <c>client_id</c> by checking it with <see cref="string.StartsWith(string)"/> 
+        /// I believe that idParts.Length > 1 will ALWAYS return true
+        /// </summary>
+        [Fact]
+        public async Task TokenEndpoint_GrantTypeClientCredentials_AsOrganization_NoIdPart_Fails()
+        {
+            var context = await _factory.Server.PostAsync("/connect/token", new FormUrlEncodedContent(new Dictionary<string, string>
+            {
+                { "grant_type", "client_credentials" },
+                { "client_id", "organization." },
+                { "client_secret", "something" },
+                { "scope", "api.organization" },
+            }));
+
+            Assert.Equal(StatusCodes.Status400BadRequest, context.Response.StatusCode);
+
+            var errorBody = await AssertHelper.AssertResponseTypeIsAsync<JsonDocument>(context);
+            var error = AssertHelper.AssertJsonProperty(errorBody.RootElement, "error", JsonValueKind.String).GetString();
+            Assert.Equal("invalid_client", error);
+        }
+
+        [Fact]
+        public async Task TokenEndpoint_GrantTypeClientCredentials_AsOrganization_OrgDoesNotExist_Fails()
+        {
+            var context = await _factory.Server.PostAsync("/connect/token", new FormUrlEncodedContent(new Dictionary<string, string>
+            {
+                { "grant_type", "client_credentials" },
+                { "client_id", $"organization.{Guid.NewGuid()}" },
+                { "client_secret", "something" },
+                { "scope", "api.organization" },
+            }));
+
+            Assert.Equal(StatusCodes.Status400BadRequest, context.Response.StatusCode);
+
+            var errorBody = await AssertHelper.AssertResponseTypeIsAsync<JsonDocument>(context);
+            var error = AssertHelper.AssertJsonProperty(errorBody.RootElement, "error", JsonValueKind.String).GetString();
+            Assert.Equal("invalid_client", error);
+        }
+
+        // This is test is commented out await confirmation from QA is our one service using this is broken or not.
```

Comment:
Did we resolve this? I'm guessing this was the push notification stuff

Reply:
Well we resolved installations being able to login but not live sync quite yet. This comment can go for now though.